frontend/src/metabase/common/components/DashboardPicker/components/DashboardPickerModal.tsx

```diff
@@ -71,7 +71,9 @@ export const DashboardPickerModal = ({
     function () {
       setSelectedItem(canSelectItem(value) ? value : null);
     },
-    [value],
+    // sync back the value to selectedItem, but only when it changes
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    [value.id, value.model],
```

Comment:
Should we use `useDeepCompareEffect` instead? 
We wouldn't need to disable eslint here.